### DISCUSSION TITLE
Offline install of sfdx-git-delta plug-in on host without any internet connection

### DISCUSSION THREAD

**Discussion by amreshvijay** [scolladon/sfdx-git-delta#376] [General]
I want to install sfdx-git-delta plug-in on a host which doesn't have an internet connection.
Is there a way to download any corresponding package for this plugin and copy over to that particular host and install it?

**Comment by mehdicherf** [comment]
Hi @amreshvijay. Thank you for sharing your request.
I'd like to understand better your use-case. 
SGD is meant to help with Salesforce deployments (which anyway requires to be connected to the Internet). 
In which situation would you need to use it on a machine that has no internet access?

**Reply by amreshvijay** [reply]
Hello @mehdisfdc - The Jenkins server has strict firewall rules applied, where it allows connection only to Salesforce, only on required port(s). It can't be open to entire internet.

but, I did a POC where I backed up following folder and files after installing the plugin, from location ~/.local/share/sfdx:
- node_modules
- package.json
- yarn.lock

After that, I uninstalled the plugin, and then restored these folder & files from backup.
And then, I could see the output of command "sfdx plugins" as:
                 sfdx-git-delta 5.5.0

So, I concluded that if I copy these folder and files to my Jenkins server, I should be able to use the plugin, without opening my Jenkins server to entire internet.

I hope, I am doing it right.
Please correct me if I'm wrong.

**Reply by mehdicherf** [reply]
Oh I see. Thank you for explaining the context.
This is not a use case we had in mind when developing the plugin. 
I believe the  plugin's code itself doesn't need access to the internet, but maybe some of the feature of the Salesforce CLI require it, I don't know. 
In other words, it could maybe work _in theory_ (I would be interest to know the result?), but it's not a supported use-case so use at own risk 😃
